### DISCUSSION TITLE
Running function via .rpc not working(local env)

### DISCUSSION THREAD

**Discussion by hilmanski** [supabase/supabase#2749] [Questions]
Hi, I have a simple function to update my timestamp like this
```
CREATE OR REPLACE FUNCTION set_timestamp(post_id uuid)
RETURNS void AS 
$$
BEGIN
    UPDATE posts 
    SET updated_at = NOW()
    WHERE id = post_id;
END;
$$ 
LANGUAGE plpgsql;
```

I tried to run it via "sql menu in supabase dahsbord" is working
time is updated
```
SELECT "set_timestamp"('141baee3-a480-41ac-9b97-9ff8c7203f97')
```

But when I use .rpc function is not
```
 const { error: errsetTimestamp } = await supabase
                .rpc('set_timestamp', { post_id })
            console.log(errsetTimestamp)
```
no error shows up, just null
but time is not updated

**Comment by hilmanski** [comment]
update: it runs on production, but not on my local dev 
weird..

**Reply by growthondev** [reply]
@hilmanski did you solve this problem?

**Reply by growthondev** [reply]
https://github.com/supabase/supabase/discussions/5913
I didn't await it.... my bad.
hope this helps someone toot :D

**Comment by silentworks** [comment]
Are you using the latest version of the Supabase client library in both environments?

**Reply by growthondev** [reply]
@silentworks I am using lastest version, but I have same result.
it works on local studio

**Comment by valvesss** [comment]
Mine work on local dev but not on production! Help me out... 

`CREATE OR REPLACE FUNCTION increment_user_bytes_usage(
    user_id UUID,
    increment_value BIGINT
) RETURNS VOID LANGUAGE plpgsql AS $$
BEGIN
    UPDATE users
    SET current_bytes_used = current_bytes_used + increment_value
    WHERE id = user_id;
END;
$$;`

Error message: File "/app/.heroku/python/lib/python3.11/site-packages/postgrest/_sync/request_builder.py", line 123, in execute
2023-10-27T22:25:51.511794+00:00 app[web.1]:     raise APIError(generate_default_error_message(r))
2023-10-27T22:25:51.511810+00:00 app[web.1]: postgrest.exceptions.APIError: {'message': 'JSON could not be generated', 'code': 204, 'hint': 'Refer to full message for details', 'details': "b''"}

Code:

`    def increment_user_bytes_usage(self, user_id: str, bytes_used: int):
        print(user_id, type(user_id))
        print(bytes_used, type(bytes_used))
        try:
            self.supabase.rpc(
                            "increment_user_bytes_usage", 
                            {"user_id": user_id, "increment_value": bytes_used}
                            ).execute()
        except APIError as error:
            logger.error(f"APIError: FAILED: increment_user_bytes_usage | Error: {error}")
            pass`

EDIT: Worked by return a JSON like object

`-- DROP FUNCTION increment_user_bytes_usage(uuid,bigint)
CREATE OR REPLACE FUNCTION increment_user_bytes_usage(
    user_id UUID,
    increment_value BIGINT
) RETURNS JSON LANGUAGE plpgsql AS $$
DECLARE
    updated_value BIGINT;
BEGIN
    UPDATE users
    SET current_bytes_used = current_bytes_used + increment_value
    WHERE id = user_id
    RETURNING current_bytes_used INTO updated_value;

    RETURN json_build_object('success', true, 'updated_bytes_used', updated_value);
END;
$$;`
